src/json/json5.cpp

```diff
@@ -144,11 +147,27 @@ std::ostream &operator<<(std::ostream &os, const JAST &jast) {
   }
 }
 
+wcl::result<JsonSubscriber, wcl::posix_error_t> JsonSubscriber::create(const char *log_path) {
+  auto res = wcl::unique_fd::open(log_path, O_APPEND | O_CREAT | O_WRONLY, 0644);
+  if (!res) {
+    return wcl::make_error<JsonSubscriber, wcl::posix_error_t>(res.error());
+  }
+  return wcl::make_result<JsonSubscriber, wcl::posix_error_t>(JsonSubscriber(std::move(*res)));
+}
+
 void JsonSubscriber::receive(const wcl::log::Event &e) {
+  std::string warning_msg = "{\"message\": \"warning: The next line may be split\"}\n";
   JAST out(JSON_OBJECT);
   for (const auto &item : e.items) {
     out.add(item.first, item.second);
   }
 
-  s << out << std::endl;
+  std::stringstream ss;
+  ss << out << std::endl;
+  std::string line = ss.str();
+
+  if (line.size() > 4095) {
+    write(to_append.get(), warning_msg.data(), warning_msg.size());
```

Comment:
this doesn't actually safely split the line unless you are saying "may be split due to interleaving"

Reply:
oh I see what you mean, you mean that I'm not adding newlines.

Reply:
I'm going to change the warning message to better reflect my intent and say "corrupted" rather than "split"